### PR TITLE
replace magic test base method

### PR DESCRIPTION
**Pull request by adamralph**
This PR does not change the behaviour of the tests aside from the timing of specific `IAsyncDocumentSession` disposals.

In `master`, disposal occurs before the test method exits for some test methods, and during NUnit tear down for other test methods. In this branch, disposal occurs before every test method exits.

It's probably easiest to review this PR one commit at a time:

- https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/c3938d8de4207dd96ec36db6ed4bf52b12120b04
- https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/9f1e4c54a4c31da9bbddbe8a76510b2a5ebebcf7 — may be easier with [white space changes hidden](https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/9f1e4c54a4c31da9bbddbe8a76510b2a5ebebcf7?w=1) due to `using` block indentation
- https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/4dfc9a746a188eaa4ffc482c30caad810d46eac4
- https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/0e373095d0ee6b1ee4dd531a5790f8a8abc1247d
- https://github.com/Particular/NServiceBus.RavenDB/pull/523/commits/3800d800c724a243b78583f4ca22720871bfb549

### CODE REVIEW

File: src/NServiceBus.RavenDB.Tests/SagaPersister/When_completing_a_version3_saga.cs [hunk 1]
```diff
@@ -16,27 +15,28 @@ public async Task Should_delete_the_unique_doc_properly()
...
+                SomeId = Guid.NewGuid()
+            };
+            var synchronizedSession = new RavenDBSynchronizedStorageSession(session);
 
-        await persister.Save(sagaEntity, this.CreateMetadata<SomeSaga>(sagaEntity), synchronizedSession, options);
+            await persister.Save(sagaEntity, this.CreateMetadata<SomeSaga>(sagaEntity), synchronizedSession, options);
 
-        //session.Advanced.GetMetadataFor(sagaEntity).Remove("NServiceBus-UniqueDocId");
-        await session.SaveChangesAsync().ConfigureAwait(false);
+            //session.Advanced.GetMetadataFor(sagaEntity).Remove("NServiceBus-UniqueDocId");
```
Comment (danielmarbach) [inline src/NServiceBus.RavenDB.Tests/SagaPersister/When_completing_a_version3_saga.cs:31]: Why not remove this?

File: src/NServiceBus.RavenDB.Tests/SagaPersister/When_persisting_a_saga_with_the_same_unique_property_as_another_saga.cs [hunk 2]
```diff
@@ -13,37 +12,40 @@ public class When_persisting_a_saga_with_the_same_unique_property_as_another_sag
...
                 UniqueString = uniqueString
             };
-            await persister.Save(saga2, this.CreateMetadata<SomeSaga>(saga2), synchronizedSession, options);
+
+            var synchronizedSession = new RavenDBSynchronizedStorageSession(session);
+
+            await persister.Save(saga1, this.CreateMetadata<SomeSaga>(saga1), synchronizedSession, options);
             await session.SaveChangesAsync().ConfigureAwait(false);
-        });
+            session.Dispose();
```
Comment (danielmarbach) [inline src/NServiceBus.RavenDB.Tests/SagaPersister/When_persisting_a_saga_with_the_same_unique_property_as_another_saga.cs:30]: Why was this left here? If I understand this test correctly the sessions don't really need to overlapp or be disposed for this test to pass right?

File: src/NServiceBus.RavenDB.Tests/SagaPersister/When_completing_a_version3_saga.cs (hunk 1 above)
Comment (adamralph) [reply src/NServiceBus.RavenDB.Tests/SagaPersister/When_completing_a_version3_saga.cs:31]: I'll add another commit to do that.

File: src/NServiceBus.RavenDB.Tests/SagaPersister/When_persisting_a_saga_with_the_same_unique_property_as_another_saga.cs (hunk 2 above)
Comment (adamralph) [reply src/NServiceBus.RavenDB.Tests/SagaPersister/When_persisting_a_saga_with_the_same_unique_property_as_another_saga.cs:30]: Good spot. I missed this line while working on the second commit. I'll fix it and amend.

**Pull request merged**